src/misc/BoundaryValueProblem/BVPSolver.cuh

```diff
@@ -42,39 +42,40 @@ namespace uammd{
   namespace BVP{
 
     namespace detail{
+      template <typename U>
       class SubsystemSolver{
 	int nz;
 	real H;
-	StorageHandle<real> CinvA_storage;
-	StorageHandle<real4> CinvABmD_storage;
+	StorageHandle<U> CinvA_storage;
+	StorageHandle<U> CinvABmD_storage;
```

Comment:
real4 was used here carefully. I understand you changed it to "U" because you have no access to something like U4.
Typically in these situations you use a construction like this:
```c++
template <typename T>
struct Real4Type;

template <>
struct Real4Type<float> {
    using Type = float4; 
};

template <>
struct Real4Type<double> {
    using Type = double4;
};
```
Something like this we could put in vector.cuh, tbh, its similar to whats being done for cufft complex types. Historically this kind of play has not been required because UAMMD typically chooses either full double or full float at compilation time. It was not designed from scratch to generate a binary that supports both at the same time. Precisely because I wanted to avoid adding "template<class T>" to every single class.
And then in the class:
```c++
template <typename U>
class SubsystemSolver {
    int nz;
    U H;  // Changed `real` to `U` here
    StorageHandle<U> CinvA_storage;

    using vec4 = typename Real4Type<U>::Type;
    StorageHandle<vec4> CinvABmD_storage;
```

Reply:
Yes, the problem was just that I don't have a U4. The solution I tried to made is something similar to what is done in BVP_complex where instead of real4 or complex4 it is used just complex and then using an array of size 4 to store the information:

`

	void registerRequiredStorage(StorageRegistration &memoryManager){

	  CinvA_storage =    memoryManager.registerStorageRequirement<complex>(2*nz+2);

	  CinvABmD_storage = memoryManager.registerStorageRequirement<complex>(4);
	}
`

Reply:
Yes, that is ok, I did not understand at first that this code intended to make BVP work with complex numbers -.-
Its all coming back to me, I wrote it in that way because this is the only place where a vec type is used and I really wanted to use thrust::complex, so it was not worth to write things like thrust::complex4.